### ISSUE TITLE
Response code/payload of AddressController REST API POST/DELETE operations

### ISSUE THREAD

**Issue by calohmn** [EnMasseProject/enmasse#1117]
Looking at the AddressController REST API, I think the response code/payload of some POST/DELETE operations should be reviewed/changed. 

***POST /addresses/{addressSpace}***
(Append one or more Addresses to the specified AddressSpace.)
Response code currently is 200 and an AddressList with all addresses is returned.

Returning all addresses here seems unnecessary and could mean a large payload if there are many addresses. 
- *In case request payload was a single address:*
I would suggest returning 201 here, with a Location header set to the /addresses/{addressSpace}/{addressName} URL and with either an empty response or returning the JSON of the single address that has been added.

- *In case the request payload was an address list:*
I think it would be better to return 201 here, with an empty response.
Alternatively, an address list with just the added addresses could be returned - that would enable to client to get to know the (sanitized) address *names* if in the request there was just the "spec.address" used.

(Having a look at the Kubernetes REST API, there is a 201 and the created resource returned on POSTing a single resource. I haven't found out about the result of POSTing a List resource, though.)

***DELETE /addressses/{addressSpace}/{address}***
(Delete an Address)
Response code currently is 200 and a List of remaining Addresses is returned.

Again, returning all addresses here seems unnecessary and could mean a large payload if there are many addresses. Therefore, I think it would be better to return code 204 with no response payload.

**Comment by calohmn** [comment]
Omitting the `listAddresses` invocation for the response would also improve processing time. 

Here's log output showing the times to add 2 addresses when there are already 6008 existing addresses:
````
2018-04-04 10:41:47.809 [vert.x-worker-thread-15] DEBUG HttpAddressService:105 - appendAddresses: [telemetry/test2_788, event/test2_788]
2018-04-04 10:41:47.818 [vert.x-worker-thread-15] DEBUG ConfigMapAddressSpaceApi:43 - getAddressSpaceWithName: get() took 8ms
[GC (Allocation Failure)  132983K->54350K(256512K), 0.0052478 secs]
2018-04-04 10:41:48.553 [vert.x-worker-thread-15] DEBUG ConfigMapAddressApi:94 - listAddresses: list() took 735ms
2018-04-04 10:41:48.586 [vert.x-worker-thread-15] DEBUG ConfigMapAddressApi:108 - createAddress: create() took 4ms
2018-04-04 10:41:48.589 [vert.x-worker-thread-15] DEBUG ConfigMapAddressApi:108 - createAddress: create() took 3ms
[GC (Allocation Failure)  132686K->61694K(243200K), 0.0076261 secs]
2018-04-04 10:41:49.109 [vert.x-worker-thread-15] DEBUG ConfigMapAddressApi:94 - listAddresses: list() took 519ms
2018-04-04 10:41:49.133 [vert.x-worker-thread-15] DEBUG HttpAddressService:48 - appendAddresses [telemetry/test2_788, event/test2_788] end (result: 6010 items) - requestProcessing took 1323ms
````

(Another improvement could be to also skip the first `listAddresses()` invocation inside `validateAddresses()` and perform individual requests to check for an already existing address - but that would require having the ".spec.address" field of the address being set in the configMap as well, so that it can be used in the labelSelector. That will be easier done once the address configMaps have been migrated to K8s resources.)

**Comment by lulf** [comment]
Fixed in 0.20.0
